File: back/domain/inputs/categoria.go
package inputs

type CategoriaInput struct {
	Nombre      string `json:"nombre"`
	Descripcion string `json:"descripcion"`
	Activo      bool   `json:"activo"`
}

type CategoriaUpdateInput struct {
	Nombre      *string `json:"nombre,omitempty"`
	Descripcion *string `json:"descripcion,omitempty"`
	Activo      *bool   `json:"activo,omitempty"`
}

File: front/src/app/models/categoria.ts
import type { Imagen } from './imagen';

export interface Categoria {
  id: number;
  nombre: string;
  descripcion: string;
  activo: boolean;
  imagenes: Imagen[];
  creado_en: string;
  actualizado_en: string;
}

export interface CategoriaInput {
  nombre: string;
  descripcion: string;
  activo: boolean;
}

export interface PaginatedCategorias {
  data: Categoria[];
  total: number;
  offset: number;
  limit: number;
}
